File: leetcode/876. Middle of the Linked List.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* middleNode(ListNode* head) {
        ListNode *current = head;
        int count = 0;
        while(current && current->next){
            count++;
            current = current->next;
        }
        //cout<< "count --> "<< count << endl;
        current = head;
        count = (count/2) + (count%2);
        for(int i = 0; i < count ; i++){
            current = current->next;
        }
        return current;
    }
};

File: online-training-platform/leetcode/876. Middle of the Linked List.kt
/**
 * Example:
 * var li = ListNode(5)
 * var v = li.`val`
 * Definition for singly-linked list.
 * class ListNode(var `val`: Int) {
 *     var next: ListNode? = null
 * }
 */
class Solution {

    private fun getSizeOfLinkedList(head: ListNode?): Int {
        var count = 0
        var current = head
        while(current != null){
            current = current?.next
            count += 1
        }
        return count
    }

    fun middleNode(head: ListNode?): ListNode? {
        val count = getSizeOfLinkedList(head); var idx = 0
        var current = head
        while(idx < count / 2){
            current = current?.next
            idx += 1
        }

        return current
    }
}